main.go

```diff
@@ -737,6 +741,45 @@ func (u *User) changeRoom(r *Room) {
 	u.room.broadcast("", Green.Paint(" --> ")+u.Name+" has joined "+Blue.Paint(u.room.name))
 }
 
+func (u *User) formatPrompt() {
+	u.FormattedPrompt = ""
+	last_escaped := false
+	for _, c := range u.Prompt {
+		if c == '\\' {
+			last_escaped = true
+		} else if last_escaped {
+			last_escaped = false
+			switch c {
+			case 'u':
+				u.FormattedPrompt += u.Name
+			case 'w':
+				u.FormattedPrompt += copyColor(u.room.name, u.Name)
+			case 'W':
+				if u.room.name == "#main" {
+					u.FormattedPrompt += copyColor("~", u.Name)
+				} else {
+					u.FormattedPrompt += copyColor("~/"+u.room.name[1:], u.Name)
+				}
+			case 't', 'T':
+				u.FormattedPrompt += fmtTime(u, time.Now())
+			case 'h', 'H':
+				u.FormattedPrompt += copyColor("devzat", u.Name)
+			case 'S':
+				u.FormattedPrompt += " "
+			default:
+				u.FormattedPrompt += string(c)
+			}
+		} else {
+			u.FormattedPrompt += string(c)
+		}
+	}
+	u.showPrompt()
+}
+
+func (u *User) showPrompt() {
```

Comment:
I think we could just inline this function (delete it and replace all its usages by that one line of code inside this function). I can do that after accepting your PR though, so this doesn't block a merge.

Reply:
Aren't you afraid this would impete readability. I know the Go compiler is not great at optimisation but this should not be too much of a problem.

Reply:
Looking at the disassembly, this call is already inlined by the compiler.

```
		u.showPrompt()
  d54f72:	90                   	nop
	u.term.SetPrompt(u.formattedPrompt)
  d54f73:	48 8b 8c 24 38 01 00 	mov    0x138(%rsp),%rcx
  d54f7a:	00 
  d54f7b:	48 8b 51 68          	mov    0x68(%rcx),%rdx
  d54f7f:	48 8b 71 20          	mov    0x20(%rcx),%rsi
  d54f83:	48 8b 79 28          	mov    0x28(%rcx),%rdi
  d54f87:	48 89 d0             	mov    %rdx,%rax
  d54f8a:	48 89 f3             	mov    %rsi,%rbx
  d54f8d:	48 89 f9             	mov    %rdi,%rcx
  d54f90:	e8 ab 75 cf ff       	callq  a4c540 <github.com/quackduck/term.(*Terminal).SetPrompt>
```

I feel like it is a good idea to have the `showPrompt` in his own function if we want to change the behavior of the prompt in the future. (I heard someone wanted to add a black line if front of it or something)

What do you think about it?

Reply:
I wasn’t really worried about the compiler output. I just thought that the function wasn’t really enhancing readability much at all, but if we expect to make further prompt changes, this is not a bad idea. Let’s keep it.